bpf/bpf_lxc.c

```diff
@@ -89,37 +153,22 @@ static __always_inline int ipv6_l3_from_lxc(struct __ctx_buff *ctx,
 	l4_off = l3_off + hdrlen;
 
 #ifndef ENABLE_HOST_SERVICES_FULL
-	{
-		struct lb6_service *svc;
-		struct lb6_key key = {};
-
-		ret = lb6_extract_key(ctx, tuple, l4_off, &key, &csum_off,
-				      CT_EGRESS);
-		if (IS_ERR(ret)) {
-			if (ret == DROP_UNKNOWN_L4)
-				goto skip_service_lookup;
-			else
-				return ret;
-		}
-
-		/*
-		 * Check if the destination address is among the address that should
-		 * be load balanced. This operation is performed before we go through
-		 * the connection tracker to allow storing the reverse nat index in
-		 * the CT entry for destination endpoints where we can't encode the
-		 * state in the address.
-		 */
-		svc = lb6_lookup_service(&key, is_defined(ENABLE_NODEPORT));
-		if (svc) {
-			ret = lb6_local(get_ct_map6(tuple), ctx, l3_off, l4_off,
-					&csum_off, &key, tuple, svc, &ct_state_new);
-			if (IS_ERR(ret))
-				return ret;
-			hairpin_flow |= ct_state_new.loopback;
-		}
+	if (!skip_lb6_local) {
+		invoke_tailcall_if(__or(__and(is_defined(ENABLE_IPV4), is_defined(ENABLE_IPV6)),
+					is_defined(DEBUG)),
+				   CILIUM_CALL_IPV6_LOCAL_FROM_LXC, tail_handle_ipv6_local);
+		return DROP_MISSED_TAIL_CALL;
```

Comment:
I don't think this is correct. `invoke_tailcall_if` is a macro that may not be compiled into a tail call, in which case we should not return `DROP_MISSED_TAIL_CALL`.

Reply:
As discussed offline, this isn't an issue because, when `invoke_tailcall_if` is not compiled into a tail call, it already calls `return`.